File: src/main/java/com/jcloisterzone/action/DonkeyAction.java
package com.jcloisterzone.action;

import com.jcloisterzone.board.Position;
import io.vavr.collection.Set;

public class DonkeyAction extends SelectTileAction {

    private final String figureId;

    public DonkeyAction(String figureId, Set<Position> options) {
        super(options);
        this.figureId = figureId;
    }

    public String getFigureId() {
        return figureId;
    }
}

File: src/main/ts/com/jcloisterzone/action/DonkeyAction.ts
import type { Set } from "../../../io/vavr/Set.js";
import type { Position } from "../board/Position.js";
import { SelectTileAction } from "./SelectTileAction.js";

/** Offer to move the donkey to another placed tile. */
export class DonkeyAction extends SelectTileAction {
  static readonly simpleName = "DonkeyAction";

  constructor(
    private readonly figureId: string,
    options: Set<Position>,
  ) {
    super(options);
  }

  getFigureId(): string {
    return this.figureId;
  }
}
